src/compute/src/server.rs

```diff
@@ -316,15 +318,15 @@ impl<'w, A: Allocate> Worker<'w, A> {
         &mut self,
         command_rx: &mut CommandReceiver,
         mut response_tx: &mut ResponseSender,
-    ) {
+    ) -> Result<(), RecvError> {
         // To initialize the connection, we want to drain all commands until we receive a
         // `ComputeCommand::InitializationComplete` command to form a target command state.
         let mut new_commands = Vec::new();
-        while let Ok(command) = command_rx.recv() {
-            if let ComputeCommand::InitializationComplete = command {
-                break;
-            } else {
-                new_commands.push(command);
+        loop {
+            match command_rx.recv() {
+                Ok(ComputeCommand::InitializationComplete) => break,
+                Ok(command) => new_commands.push(command),
+                Err(e) => return Err(e),
```

Comment:
Could this be simpler as `match command_rx.recv()?`?

Reply:
Oh yes! Thanks for pointing it out.